File: VRPortalNexGen/ClientApp/src/app/Models/data-source-column-list.ts
import {ColumnList} from '../Models/column-list';
import {Spparameter} from '../Models/spparameter';
import { DataTableFilter } from './data-table-filter';
import { DBDetails } from './dbdetails';


export class DatabaseNameList {
  dbName: string;
}
export class TableNameList {
  tableName: string;
}

export class StoredProcedureNameList {
  storedProcedureName: string;
}
export class StoredProcedureParameterNameList {
  storedProcedureParameterName: string;
}

export class Flags {
  public IsDBAvailable = false;
  public IsDatabaseSelected = false;
  public IsTableAvailable = false;
  public IsStoredProcedureAvailable = false;
  public IsQuerySelected = false;
  public IsColumnAvailable = false;
  public IsResultAvailable = false;
  public show = false;
  public windowsflag = false;
  public sqlflag = false;
}

export class DataSourceColumnList {
  constructor() {
    this.Flag = new Flags();
    this.ColumnNameList = [];
  }
  DataSourceType: string;
  Source: string;
  FileName: string;
  FilePath: string;
  CommonColumn: string;
  ColumnNameList: ColumnList[];
  RequestType: string;
  SourceFile: File;
  DBServerName: string;
  DBName: string;
  TableName: string;
  ColumnName: string;
  AuthenticationType: string;
  UserName: string;
  Password: string;
  DatabaseObjectType: string;
  StoredProcedureName: string;
  SpParameterValue: any = {};
  DataTableFilterList: DataTableFilter[];
  DBDetailsObj: DBDetails;
  dbNameList: DatabaseNameList[];
  tableNameList: TableNameList[];
  storedProcedureNameList: StoredProcedureNameList[];
  storedProcedureParameterNameList: StoredProcedureParameterNameList[];
  SqlQuery: string;
  Flag: Flags;
}

File: VRPortal.Models/DataSourceColumnList.cs
using System;
using System.Collections.Generic;

using Microsoft.Azure.Cosmos.Table;

namespace VRPortal.Models
{
    public class DataSourceColumnList : TableEntity
    {
        public DataSourceColumnList()
        {

        }

        public DataSourceColumnList(string partitionKey, string rowKey)
        {
            PartitionKey = partitionKey;
            RowKey = rowKey;
        }

        public string DataSourceType { get; set; }
        public string Source { get; set; }
        public string FileName { get; set; }
        public string FilePath { get; set; }
        public string TableName { get; set; }
        public string SheetName { get; set; }

        public string ColumnName { get; set; }
        public List<DataColumns> ColumnNameList { get; set; }
        public string CommonColumn { get; set; }
        public string DateFilterColumnName { get; set; }
        public string DateFilterColumnValue { get; set; }
        public List<DataTableFilter> DataTableFilterList { get; set; }

        public string DbServerName { get; set; }
        public string DBName { get; set; }
        public string SqlQuery { get; set; }
        public string DatabaseObjectType { get; set; }
        public string StoredProcedureName { get; set; }
        public Dictionary<string, string> SpParameterValue { get; set; }
        public string UserName { get; set; }
        public string Password { get; set; }
        public string AuthenticationType { get; set; }
    }
}